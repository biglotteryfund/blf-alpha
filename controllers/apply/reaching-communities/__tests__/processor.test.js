/* eslint-env jest */
'use strict';
const nodemailer = require('nodemailer');

const processor = require('../processor');
const form = require('../form-model');
const { flattenFormData, stepsWithValues } = require('../../helpers');

const mockFormData = {
    'step-1': {
        'your-idea': 'Test idea'
    },
    'step-2': {
        location: 'South West',
        'project-location': 'London'
    },
    'step-3': {
        'organisation-name': 'Test Organisation',
        'additional-organisations': ''
    },
    'step-4': {
        'first-name': 'Example',
        'last-name': 'Person',
        email: 'example@example.com',
        'phone-number': '03454102030'
    }
};

function cleanMailForSnaphot(info) {
    info.message = info.message
        .toString()
        .replace(/Message-ID: .*\n/, '')
        .replace(/Date: .*\n/, '')
        .replace(/ boundary=.*\n/, '')
        .replace(/----.*\n/gm, '');
    return info;
}

describe('processor', () => {
    it('should work', async () => {
        const transport = nodemailer.createTransport({
            streamTransport: true,
            newline: 'unix',
            buffer: true
        });

        const result = await processor(
            {
                form: form,
                data: flattenFormData(mockFormData),
                stepsWithValues: stepsWithValues(form.steps, mockFormData)
            },
            transport
        );

        const internalEmail = cleanMailForSnaphot(result[0]);
        const externalEmail = cleanMailForSnaphot(result[1]);

        expect(internalEmail.message).toMatchSnapshot();
        expect(externalEmail.message).toMatchSnapshot();
    });
});
