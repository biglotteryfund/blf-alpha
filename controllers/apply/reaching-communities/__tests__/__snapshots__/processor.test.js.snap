// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`processor should work 1`] = `
"Content-Type: multipart/alternative;
From: Big Lottery Fund <noreply@biglotteryfund.org.uk>
To: Example Person <example@example.com>
Subject: Thank you for getting in touch with the Big Lottery Fund!
MIME-Version: 1.0

Content-Type: text/plain; charset=utf-8
Content-Transfer-Encoding: quoted-printable

Thank you for getting in touch with us.

You will see a copy of the =
information you submitted below. This will be forwarded to one of our =
funding officers who will get in
touch within fifteen working days to let =
you know if we can help.

In the meantime, you can find out more about our =
funding and how we can support your community on our website:=20
www.biglotteryfund.org.uk/england [https://www.biglotteryfund.org.=
uk/england]

Thanks,

The Big Lottery Fund

YOUR INFORMATION
YOUR IDEA
BRIEFLY EXPLAIN YOUR IDEA AND WHY IT=E2=80=99LL MAKE A DIFFERENCE
Test idea

PROJECT LOCATION
SELECT ALL REGIONS THAT APPLY
South West

PROJECT LOCATION
London

YOUR ORGANISATION
LEGAL NAME
Test Organisation

YOUR DETAILS
FIRST NAME
Example

LAST NAME
Person

EMAIL ADDRESS
example@example.com

PHONE NUMBER
03454102030

DATA PROTECTION
The Big Lottery Fund is a public body with a duty to distribute National =
Lottery and other money in grants for good causes. We use
the personal data you provide, such as contact details for individuals at =
your organisation, to help you apply for a grant and to
assess your application. We may share your personal data with organisations=
 which help us with our grant making activities or
others which have a =
legitimate interest in our work or have funded your grant. We will only =
share personal data which they need to
carry out their work and subject to =
appropriate safety measures. Our Data Protection and Privacy Notice gives =
more information
about how we store and use personal data and the lawful =
basis for this. Please read the full Notice which is published on our
website at www.biglotteryfund.org.uk/data-protection or contact us to =
request a hard copy. The Notice may be updated from time to
time.
Content-Type: text/html; charset=utf-8
Content-Transfer-Encoding: quoted-printable


<!doctype html>
<html>
    <head>
       =20
        <style>
@font-face {
  font-family: 'Poppins';
  font-style: normal;
  font-weight: 300;
  src: local('Poppins Light'), local('Poppins-Light'), url(https://fonts.=
gstatic.com/s/poppins/v5/pxiByp8kv8JHgFVrLDz8Z1xlEA.ttf) =
format('truetype');}
@font-face {
  font-family: 'Poppins';
  font-style: normal;
  font-weight: 500;
  src: local('Poppins Medium'), =
local('Poppins-Medium'), url(https://fonts.gstatic.com/s/poppins/v5/pxiByp8=
kv8JHgFVrLGT9Z1xlEA.ttf) format('truetype');}
</style>
        <meta charset=3D\\"UTF-8\\">
    </head>
    <body style=3D\\"margin: =
auto; max-width: 75%; padding: 20px; color: #333; font-family: arial; =
font-size: 20px; font-weight: 400; line-height: 1.35;\\">
        <div class=3D\\"logo\\" style=3D\\"display: block; margin-bottom: 20px; =
text-align: right;\\">
            <img src=3D\\"https://www.biglotteryfund.org=
.uk/assets/images/logos/logo.png\\" alt=3D\\"Big Lottery Fund logo\\" =
style=3D\\"width: 150px;\\" width=3D\\"150\\">
        </div>

        <p style=3D\\"color: #333; font-family: arial; font-size: 20px; =
font-weight: 400; line-height: 1.35;\\">Thank you for getting in touch with =
us.</p>

        <p style=3D\\"color: #333; font-family: arial; font-size: =
20px; font-weight: 400; line-height: 1.35;\\">You will see a copy of the =
information you submitted below. This will be forwarded to one of our =
funding officers who will get in touch within fifteen working days to let =
you know if we can help.</p>
        <p style=3D\\"color: #333; font-family: =
arial; font-size: 20px; font-weight: 400; line-height: 1.35;\\">In the =
meantime, you can find out more about our funding and how we can support =
your community on our website: <a href=3D\\"https://www.biglotteryfund.org.=
uk/england\\" style=3D\\"color: #365cc9;\\">www.biglotteryfund.org.=
uk/england</a></p>
        <p style=3D\\"color: #333; font-family: arial; =
font-size: 20px; font-weight: 400; line-height: 1.35;\\">Thanks,</p>
        <p style=3D\\"color: #333; font-family: arial; font-size: 20px; =
font-weight: 400; line-height: 1.35;\\"><strong>The Big Lottery =
Fund</strong></p>

        <h2 class=3D\\"space-above\\" style=3D\\"color: =
#e20087; font-weight: 500; margin: 0 0 20px; font-size: 22px; margin-top: =
50px;\\">Your information</h2>
        <div class=3D\\"summary\\">
           =20
   =20
        <h4 style=3D\\"color: #e20087; margin: 0 0 20px; font-size: =
16px; border-bottom: 1px solid #e20087; padding-bottom: 3px; font-weight: =
700;\\">Your idea</h4>
       =20
           =20
               =20
                    <h5 style=3D\\"color: #e20087; margin: 0 0 20px; =
font-size: 14px; font-weight: 700;\\">Briefly explain your idea and why =
it=E2=80=99ll make a difference</h5>
                   =20
                        <p style=3D\\"color: #333; font-family: arial; =
font-size: 20px; font-weight: 400; line-height: 1.35;\\">Test idea</p>
                   =20
               =20
           =20
       =20
   =20
        <h4 style=3D\\"color: #e20087; margin: 0 0 20px; font-size: 16px; =
border-bottom: 1px solid #e20087; padding-bottom: 3px; font-weight: =
700;\\">Project location</h4>
       =20
           =20
               =20
                    <h5 style=3D\\"color: #e20087; margin: 0 0 20px; =
font-size: 14px; font-weight: 700;\\">Select all regions that apply</h5>
                   =20
                        <p style=3D\\"color: #333; =
font-family: arial; font-size: 20px; font-weight: 400; line-height: 1.=
35;\\">South West</p>
                   =20
               =20
           =20
               =20
                    <h5 style=3D\\"color: =
#e20087; margin: 0 0 20px; font-size: 14px; font-weight: 700;\\">Project =
location</h5>
                   =20
                        <p =
style=3D\\"color: #333; font-family: arial; font-size: 20px; font-weight: =
400; line-height: 1.35;\\">London</p>
                   =20
               =20
           =20
       =20
   =20
        <h4 style=3D\\"color: #e20087; margin: 0 0 20px; font-size: 16px; =
border-bottom: 1px solid #e20087; padding-bottom: 3px; font-weight: =
700;\\">Your organisation</h4>
       =20
           =20
               =20
                    <h5 style=3D\\"color: #e20087; margin: 0 0 20px; =
font-size: 14px; font-weight: 700;\\">Legal name</h5>
                   =20
                        <p style=3D\\"color: #333; font-family: arial; =
font-size: 20px; font-weight: 400; line-height: 1.35;\\">Test =
Organisation</p>
                   =20
               =20
           =20
               =20
           =20
       =20
   =20
        <h4 style=3D\\"color: #e20087; margin: 0 0 20px; font-size: 16px; =
border-bottom: 1px solid #e20087; padding-bottom: 3px; font-weight: =
700;\\">Your details</h4>
       =20
           =20
               =20
                    <h5 style=3D\\"color: #e20087; margin: 0 0 20px; =
font-size: 14px; font-weight: 700;\\">First name</h5>
                   =20
                        <p style=3D\\"color: #333; font-family: arial; =
font-size: 20px; font-weight: 400; line-height: 1.35;\\">Example</p>
                   =20
               =20
           =20
               =20
                    <h5 style=3D\\"color: #e20087; margin: 0 0 20px; =
font-size: 14px; font-weight: 700;\\">Last name</h5>
                   =20
                        <p style=3D\\"color: #333; font-family: arial; =
font-size: 20px; font-weight: 400; line-height: 1.35;\\">Person</p>
                   =20
               =20
           =20
               =20
                    <h5 style=3D\\"color: #e20087; margin: 0 0 20px; =
font-size: 14px; font-weight: 700;\\">Email address</h5>
                   =20
                        <p style=3D\\"color: #333; =
font-family: arial; font-size: 20px; font-weight: 400; line-height: 1.=
35;\\">example@example.com</p>
                   =20
               =20
           =20
               =20
                    <h5 style=3D\\"color: =
#e20087; margin: 0 0 20px; font-size: 14px; font-weight: 700;\\">Phone =
number</h5>
                   =20
                        <p =
style=3D\\"color: #333; font-family: arial; font-size: 20px; font-weight: =
400; line-height: 1.35;\\">03454102030</p>
                   =20
               =20
           =20
       =20
   =20

        </div>

       =20

        <h5 class=3D\\"space-above\\" style=3D\\"color: #e20087; =
font-weight: 500; margin: 0 0 20px; font-size: 14px; margin-top: =
50px;\\">Data protection</h5>
        <small style=3D\\"color: #666; display: =
block; font-size: 11px; margin-bottom: 10px;\\">The Big Lottery Fund is a =
public body with a duty to distribute National Lottery and other money in =
grants for good causes. We use the personal data you provide, such as =
contact details for individuals at your organisation, to help you apply for=
 a grant and to assess your application.</small>
        <small =
style=3D\\"color: #666; display: block; font-size: 11px; margin-bottom: =
10px;\\">We may share your personal data with organisations which help us =
with our grant making activities or others which have a legitimate interest=
 in our work or have funded your grant. We will only share personal data =
which they need to carry out their work and subject to appropriate safety =
measures.</small>
        <small style=3D\\"color: #666; display: block; =
font-size: 11px; margin-bottom: 10px;\\">Our Data Protection and Privacy =
Notice gives more information about how we store and use personal data and =
the lawful basis for this. Please read the full Notice which is published =
on our website at www.biglotteryfund.org.uk/data-protection or contact us =
to request a hard copy. The Notice may be updated from time to time.=
</small>

    </body>
</html>

"
`;

exports[`processor should work 2`] = `
"Content-Type: multipart/alternative;
From: Big Lottery Fund <noreply@biglotteryfund.org.uk>
To: Example Person <example@example.com>
Subject: New idea submission from website: Test Organisation
MIME-Version: 1.0

Content-Type: text/plain; charset=utf-8
Content-Transfer-Encoding: quoted-printable

A customer has submitted an idea from the Reaching Communities & =
Partnerships web form.

The details they provided are included below:

CUSTOMER INFORMATION
YOUR DETAILS
FIRST NAME
Example

LAST NAME
Person

EMAIL ADDRESS
example@example.com

PHONE NUMBER
03454102030

YOUR ORGANISATION
LEGAL NAME
Test Organisation

PROJECT LOCATION
SELECT ALL REGIONS THAT APPLY
South West

PROJECT LOCATION
London

YOUR IDEA
BRIEFLY EXPLAIN YOUR IDEA AND WHY IT=E2=80=99LL MAKE A DIFFERENCE
Test idea
Content-Type: text/html; charset=utf-8
Content-Transfer-Encoding: quoted-printable


<!doctype html>
<html>
    <head>
       =20
        <style>
@font-face {
  font-family: 'Poppins';
  font-style: normal;
  font-weight: 300;
  src: local('Poppins Light'), local('Poppins-Light'), url(https://fonts.=
gstatic.com/s/poppins/v5/pxiByp8kv8JHgFVrLDz8Z1xlEA.ttf) =
format('truetype');}
@font-face {
  font-family: 'Poppins';
  font-style: normal;
  font-weight: 500;
  src: local('Poppins Medium'), =
local('Poppins-Medium'), url(https://fonts.gstatic.com/s/poppins/v5/pxiByp8=
kv8JHgFVrLGT9Z1xlEA.ttf) format('truetype');}
</style>
        <meta charset=3D\\"UTF-8\\">
    </head>
    <body style=3D\\"margin: =
auto; max-width: 75%; padding: 20px; color: #333; font-family: arial; =
font-size: 20px; font-weight: 400; line-height: 1.35;\\">

        <p style=3D\\"color: #333; font-family: arial; font-size: 20px; =
font-weight: 400; line-height: 1.35;\\">A customer has submitted an idea from=
 the Reaching Communities &amp; Partnerships web form.</p>
        <p style=3D\\"color: #333; font-family: arial; font-size: 20px; =
font-weight: 400; line-height: 1.35;\\">The details they provided are =
included below:</p>

        <h2 class=3D\\"space-above\\" style=3D\\"color: =
#e20087; font-weight: 500; margin: 0 0 20px; font-size: 22px; margin-top: =
50px;\\">Customer information</h2>
        <div class=3D\\"summary\\">
           =20
   =20
        <h4 style=3D\\"color: #e20087; margin: 0 0 =
20px; font-size: 16px; border-bottom: 1px solid #e20087; padding-bottom: =
3px; font-weight: 700;\\">Your details</h4>
       =20
           =20
               =20
                    <h5 style=3D\\"color: #e20087; margin:=
 0 0 20px; font-size: 14px; font-weight: 700;\\">First name</h5>
                   =20
                        <p style=3D\\"color: #333; =
font-family: arial; font-size: 20px; font-weight: 400; line-height: 1.=
35;\\">Example</p>
                   =20
               =20
           =20
               =20
                    <h5 style=3D\\"color: #e20087; margin:=
 0 0 20px; font-size: 14px; font-weight: 700;\\">Last name</h5>
                   =20
                        <p style=3D\\"color: #333; =
font-family: arial; font-size: 20px; font-weight: 400; line-height: 1.=
35;\\">Person</p>
                   =20
               =20
           =20
               =20
                    <h5 style=3D\\"color: #e20087; margin:=
 0 0 20px; font-size: 14px; font-weight: 700;\\">Email address</h5>
                   =20
                        <p style=3D\\"color: #333; =
font-family: arial; font-size: 20px; font-weight: 400; line-height: 1.=
35;\\">example@example.com</p>
                   =20
               =20
           =20
               =20
                    <h5 style=3D\\"color: =
#e20087; margin: 0 0 20px; font-size: 14px; font-weight: 700;\\">Phone =
number</h5>
                   =20
                        <p =
style=3D\\"color: #333; font-family: arial; font-size: 20px; font-weight: =
400; line-height: 1.35;\\">03454102030</p>
                   =20
               =20
           =20
       =20
   =20
        <h4 style=3D\\"color: #e20087; margin: 0 0 20px; font-size: 16px; =
border-bottom: 1px solid #e20087; padding-bottom: 3px; font-weight: =
700;\\">Your organisation</h4>
       =20
           =20
               =20
                    <h5 style=3D\\"color: #e20087; margin: 0 0 20px; =
font-size: 14px; font-weight: 700;\\">Legal name</h5>
                   =20
                        <p style=3D\\"color: #333; font-family: arial; =
font-size: 20px; font-weight: 400; line-height: 1.35;\\">Test =
Organisation</p>
                   =20
               =20
           =20
               =20
           =20
       =20
   =20
        <h4 style=3D\\"color: #e20087; margin: 0 0 20px; font-size: 16px; =
border-bottom: 1px solid #e20087; padding-bottom: 3px; font-weight: =
700;\\">Project location</h4>
       =20
           =20
               =20
                    <h5 style=3D\\"color: #e20087; margin: 0 0 20px; =
font-size: 14px; font-weight: 700;\\">Select all regions that apply</h5>
                   =20
                        <p style=3D\\"color: #333; =
font-family: arial; font-size: 20px; font-weight: 400; line-height: 1.=
35;\\">South West</p>
                   =20
               =20
           =20
               =20
                    <h5 style=3D\\"color: =
#e20087; margin: 0 0 20px; font-size: 14px; font-weight: 700;\\">Project =
location</h5>
                   =20
                        <p =
style=3D\\"color: #333; font-family: arial; font-size: 20px; font-weight: =
400; line-height: 1.35;\\">London</p>
                   =20
               =20
           =20
       =20
   =20
        <h4 style=3D\\"color: #e20087; margin: 0 0 20px; font-size: 16px; =
border-bottom: 1px solid #e20087; padding-bottom: 3px; font-weight: =
700;\\">Your idea</h4>
       =20
           =20
               =20
                    <h5 style=3D\\"color: #e20087; margin: 0 0 20px; =
font-size: 14px; font-weight: 700;\\">Briefly explain your idea and why =
it=E2=80=99ll make a difference</h5>
                   =20
                        <p style=3D\\"color: #333; font-family: arial; =
font-size: 20px; font-weight: 400; line-height: 1.35;\\">Test idea</p>
                   =20
               =20
           =20
       =20
   =20

        </div>

    </body>
</html>

"
`;
